src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1519,14 +1519,8 @@ function fillcombinations(df::AbstractDataFrame, indexcols;
     # Create a vector of vectors of unique values in each column
     uniquevals = []
     for col in colind
-        # levels drops missing, handle the case where missing values are present
         # All levels are retained, missing is added only if present
-        # TODO: change this after DataAPI.jl levels supports missing
-        if any(ismissing, df[!, col])
-            tempcol = vcat(levels(df[!, col]), missing)
-        else
-            tempcol = levels(df[!, col])
-        end
+        tempcol = levels(df[!, col], skipmissing=false)
```

Comment:
I made this change in this PR since we bump DataAPI.jl version in it. Alternatively I can later open another PR for only this change.